### ISSUE TITLE
Expose histogram through the Prometheus endpoint

### ISSUE THREAD

**Issue by kallekrantz** [Unleash/unleash#764]
When deploying unleash, I wanted to use some the rich tooling around Prometheus Histograms that exists in Grafana.

However, Unleash currently only exposes the summaries, this makes it really difficult to aggregate the data across different dimensions (such as checking averaging all across all status codes) which we'd find really useful.

I understand that it would be a breaking change to change the metric type, but maybe making it configurable is enough, or maybe even exposing both?

https://prometheus.io/docs/practices/histograms/#quantiles

**Comment by ivarconr** [comment]
Yes Summaries is kind of a compromise. It gives you the percentiles for less computational cost (and less to do for the user), per metric per instance. There is no way to mathematically do anything meaningful across different summaries. 

Generally we have found that having the summaries individually has proven a good enough, given that you just use it for monitoring and alerting. 

Even if you aggregate metrics, you should still look at them individually: 
- E.g. because of how Unleash works the `/api/client/features` will have the most traffic and will yield very low response times. This will influence the totals if you start aggregating. 
- If one instance in starting to slow down is good to detect this early, even if other instances are performing like they should. 

In general I agree that Histograms are more mathematically correct. But at least for us hosting hundreds of instances of many years summarizes have provided a good enough approximation, as we generally want to alert on individual metrics series.

I think we can make it configurable, and possible provide both (given that the cost of this is acceptable).
